check/checksum.go

```diff
@@ -0,0 +1,182 @@
+package check
+
+import (
+	"bytes"
+	"crypto/md5"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/schollz/progressbar/v3"
+)
+
+func isHiddenFile(f string) bool {
+	return string(f[0]) == "."
+}
+
+func checksumFor(path string) ([]byte, error) {
+	h := md5.New()
+
+	var r *os.File
+	r, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("error opening file in %s: %w", path, err)
+	}
+	defer r.Close()
+
+	if _, err := io.Copy(h, r); err != nil {
+		return nil, fmt.Errorf("error reading %s content: %w", path, err)
+	}
+
+	return h.Sum(nil), nil
+}
+
+func checkChecksum(src, target string) (bool, error) {
+	name := filepath.Base(src)
+	sc, err := checksumFor(src)
+	if err != nil {
+		return false, fmt.Errorf("error getting %s checksum: %w", src, err)
+	}
+
+	targetPath := filepath.Join(target, name)
+	tc, err := checksumFor(targetPath)
+	if err != nil {
+		return false, fmt.Errorf("error getting %s checksum: %w", targetPath, err)
+	}
```

Comment:
Apenas sugerindo variáveis com nomes mais curtos, que é o estilo do Go

```suggestion
	pth := filepath.Join(target, name)
	tc, err := checksumFor(pth)
	if err != nil {
		return false, fmt.Errorf("error getting %s checksum: %w", pth, err)
	}
```